linkpreview.go

```diff
@@ -1,17 +1,38 @@
 package linkpreview
 
 import (
+	"errors"
 	"io"
 	"net/http"
+	"net/url"
+	"strings"
 
 	"github.com/PuerkitoBio/goquery"
 )
 
+func validateURL(link string) error {
+	parsedURL, err := url.ParseRequestURI(link)
+	if err != nil || !strings.HasPrefix(parsedURL.Scheme, "http") {
+		return errors.New("invalid URL")
+	}
+	return nil
+}
+
 func Parse(link string, options ...ConfigOption) (ParserResult, error) {
+	if err := validateURL(link); err != nil {
+		return ParserResult{}, err
+	}
+
 	request, err := http.NewRequest("GET", link, nil)
 	if err != nil {
 		return ParserResult{}, err
 	}
+
+	// Add security headers
+	request.Header.Set("X-Content-Type-Options", "nosniff")
+	request.Header.Set("X-Frame-Options", "DENY")
+	request.Header.Set("X-XSS-Protection", "1; mode=block")
```

Comment:
I think this is incorrect usage of secure headers. They are for response headers only.
Please see: https://cheatsheetseries.owasp.org/cheatsheets/HTTP_Headers_Cheat_Sheet.html